### PR TITLE
Update ensure_tf_install.py

### PR DESCRIPTION
**Pull request by seanpmorgan**
# Description

Update the maximum TF version
Part of https://github.com/tensorflow/addons/issues/2075

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [ ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

File: tensorflow_addons/utils/ensure_tf_install.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 import tensorflow as tf
 
 MIN_TF_VERSION = "2.2.0"
-MAX_TF_VERSION = "2.3.0"
+MAX_TF_VERSION = "2.4.0"
```
Comment (bhack) [inline tensorflow_addons/utils/ensure_tf_install.py:27]: So does this mean that we are compatibile with 2.3.x for the upstream version policy?

Comment (seanpmorgan) [reply tensorflow_addons/utils/ensure_tf_install.py:27]: Yes it should. TF patch version should have no breaking changes. I'm open to restructuring how this is written per your comment in the issue but for now this is how it was previously set up and I'd like to fix our broken package

Comment (bhack) [reply tensorflow_addons/utils/ensure_tf_install.py:27]: If we can trust upstream versioning contract without our CI testing It Is ok for me

Comment (WindQAQ) [reply tensorflow_addons/utils/ensure_tf_install.py:27]: I think it should be ok for this.

Comment (bhack) [reply tensorflow_addons/utils/ensure_tf_install.py:27]: Ok but you need to be aware that `tf.experimental` is not in the contract for minor versions and we have already some experimental call in TFA (we could add other dependecies on experimental in the future):

See https://www.tensorflow.org/guide/versions?hl=en#what_is_not_covered

**Pull request merged**